scripts/invalidCompartment/invalidCompartment.ps1

```diff
@@ -0,0 +1,64 @@
+function Is-CompartmentIdValid() {
+	Write-Host "#===== Validating Compartment ID in Endpoint and Namespace =====#"
+	Write-Host ""
+	$eps = Get-HnsEndpoint
+	$nss = Get-HnsNamespace
+
+	foreach($ep in $eps) {
+		if ($ep.Resources.Allocators.CompartmendId.Length -gt 0) {
+			$compId = [int]$($ep.Resources.Allocators.CompartmendId[0])
+			$nsId = $ep.Namespace.ID
+			foreach($ns in $nss) {
+				if ($ns.ID -Eq $nsId) {
+					Write-Host "IP Address : $($ep.IPAddress), NamespaceID: $nsId, EPCompID : $compId, NSCompID: $($ns.CompartmentId)"	
+					if ($ns.CompartmentId -NE $compId) {
+						Write-Host "CompartmentId not matching..."
+						return $false
+					} else {
+						Write-Host "CompartmentId matching..."
+					}
+				}
+			}
+		}
+	}
+	Write-Host ""
+	return $true
+}
+
+$iter = 1
+$curLoc = (Get-Location).Path
+pktmon stop # Stopping if pktmon is already running
+
+# Start pktmon
+Write-Host "#===== Starting pktmon with trace level 6 for TCPIP and Host-Network-Service =====#"
+pktmon start --capture --comp all --trace -p Microsoft-Windows-TCPIP -k 0xFF -l 6 -p Microsoft-Windows-Host-Network-Service -l 6 -f traces.etl --pkt-size 100 -s 1024
+
+
+While(Is-CompartmentIdValid) {
+	$d = Get-Date
+	Write-Host "#===== Iteration: $iter completed at $d. No issue found. Waiting for 10 minutes for next iteration. =====#"
+	Write-Host ""
+	Start-Sleep -Seconds 600
+	$iter++
+}
+
+Write-Host "#===== Issue detected. Stopping pktmon and collecting logs. =====#"
+Write-Host ""
+# Stop pktmon
+pktmon stop
+
+# Collecting Windows logs
+C:\k\debug\collect-windows-logs.ps1
+
+Write-Host "Traces available in $curLoc\traces.etl"
+
+# The below While loop will keep the script running indefinitely keeping the hpc pod alive.

```

Comment:
Can we keep it only for the POD yaml, and not on the script

Reply:
Better to have all business logic in one place which is script.